pkg/manager/manager.go

```diff
@@ -36,22 +36,22 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/leaderelection"
 	"sigs.k8s.io/controller-runtime/pkg/metrics"
 	"sigs.k8s.io/controller-runtime/pkg/recorder"
+	"sigs.k8s.io/controller-runtime/pkg/inject"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
 )
 
 // Manager initializes shared dependencies such as Caches and Clients, and provides them to Runnables.
 // A Manager is required to create Controllers.
 type Manager interface {
+	// Every manager knows how to inject dependencies into other things.
+	inject.Context
```

Comment:
I'm wondering if changing it to `Dependency` or `Dependencies` makes more sense.
So this user-facing piece becomes `inject.Dependency` or `inject.Dependencies`.